plugins/text-freetype2/text-functionality.c

```diff
@@ -141,10 +141,55 @@ void set_up_vertex_buffer(struct ft2_source *srcdata)
 	}
 
 skip_word_wrap:;
+
+	// calculate lines width for alignment
+	len = wcslen(srcdata->text);
+	uint32_t current_line_width = 0;
+	uint32_t current_line_index = 0;
+	if (srcdata->custom_width < 100) {
+		srcdata->max_line_width = 0;
+	} else {
+		srcdata->max_line_width = srcdata->custom_width;
+	}
+	for (uint32_t i = 0; i <= len; i++) {
+		if (srcdata->text[i] == L'\n' || i == len) {
+			if (current_line_width > srcdata->max_line_width) {
+				srcdata->max_line_width = current_line_width;
+			}
+			if (current_line_index < max_aligned_lines) {
+				srcdata->lines_width[current_line_index] = current_line_width;
+				current_line_index++;
+			}
+			current_line_width = 0;
+		} else {
+			glyph_index = FT_Get_Char_Index(srcdata->font_face,
+				srcdata->text[i]);
+			current_line_width += src_glyph->xadv;
+		}
+	}
+
 	fill_vertex_buffer(srcdata);
 	obs_leave_graphics();
 }
 
+uint32_t get_dx_for_line(struct ft2_source *srcdata, int line_index) {
+	if (line_index >= max_aligned_lines) {
+		return 0;
+	}
+
+	switch (srcdata->alignment) {
+		case TEXT_ALIGNMENT_CENTER: {
+			return (srcdata->max_line_width - srcdata->lines_width[line_index]) / 2;
+		}
+		case TEXT_ALIGNMENT_RIGHT: {
+			return srcdata->max_line_width - srcdata->lines_width[line_index];
+		}
+		default: {
+			return 0;
+		}
+	}
+}
```

Comment:
We start the brace on the next line after functions.  And some of these lines go over the column limit.  `case` statements should be on the same column as the switch.  Additionally, the braces for the case statements feel a bit unnecessary because you're not really declaring any variables there.  This block should be:

```C
uint32_t get_dx_for_line(struct ft2_source *srcdata, int line_index)
{
	if (line_index >= max_aligned_lines) {
		return 0;
	}

	switch (srcdata->alignment) {
	case TEXT_ALIGNMENT_CENTER:
		return (srcdata->max_line_width -
				srcdata->lines_width[line_index]) / 2;
	case TEXT_ALIGNMENT_RIGHT:
		return srcdata->max_line_width -
			srcdata->lines_width[line_index];
	default:
		return 0;
	}
}
```